File: src/1-500/055/JumpGame.py
from typing import List


class Solution:
    def canJump(self, nums: List[int]) -> bool:
        reach = 0
        for i, num in enumerate(nums):
            if i > reach:
                return False
            reach = max(reach, i + num)
        return True


solution = Solution()
print(solution.canJump([2,3,1,1,4]))

File: src/1-500/055/JumpGame.java
class Solution {
    public boolean canJump(int[] nums) {
        if (nums == null || nums.length == 0) {
            return true;
        }
        int maxRight = 0;
        for (int i = 0; i < nums.length; i++) {
            if (maxRight < i) {
                return false;
            }
            maxRight = Math.max(i + nums[i], maxRight);
            if (maxRight >= nums.length - 1) {
                return true;
            }
        }
        return false;
    }
}